addons/website/static/src/js/editor/widget_link.js

```diff
@@ -40,29 +43,71 @@ weWidgets.LinkTools.include({
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @override web_editor - link.js abstract method
+     * @param {string} url
+     * @returns {boolean} Whether or not link should open in a new page
+     * @private
+     */
+    _isNewWindow: function (url) {
+        return url && url.includes('download');
+    },
+    /**
+     * @private
+     */
+     _onLinkUploadFileClick: function () {
+        const restoreSelection = () => {
+            this.$link.click();
+            this.$button.click();
+        };
+        const documentDialog = new weWidgets.MediaDialog(this, {
+            noIcons: true,
+            noImages: true,
+            noVideos: true,
+        });
+
+        documentDialog.on('save', this, function (htmlOutput) {
+            const elem = $(htmlOutput);
+            let url;
+            if (elem.is('a')) {
+                url = elem.attr('href');
+            } else {
+                url = `${elem.attr('src')}?download=true`;
+            }
+            this.$urlInput.val(url).trigger('input');
+            restoreSelection();
+        });
+        documentDialog.on('closed', this, function () {
+            if (documentDialog.destroyAction !== 'save') {
+                restoreSelection();
+            }
```

Comment:
```suggestion
            restoreSelection();
```
As remove `restoreSelection();` in the `save` 3 lines above as save is calling closed, right?